_15thnight/__init__.py

```diff
@@ -4,44 +4,63 @@
 from flask.ext.login import LoginManager, current_user, logout_user
 from werkzeug.datastructures import MultiDict
 
-from _15thnight import database, queue
-from _15thnight.api import (
-    account_api, alert_api, category_api, need_api, response_api, service_api,
-    user_api
-)
 from _15thnight.email import mailer
 from _15thnight.forms import csrf_protect
 from _15thnight.models import User
 from _15thnight.util import ExtensibleJSONEncoder
 
 
-app = Flask(__name__)
-
-try:
-    app.config.from_object('config')
-except:
-    app.config.from_object('configdist')
-
-app.secret_key = app.config['SECRET_KEY']
-app.json_encoder = ExtensibleJSONEncoder
-
-app.register_blueprint(alert_api, url_prefix='/api/v1')
-app.register_blueprint(account_api, url_prefix='/api/v1')
-app.register_blueprint(category_api, url_prefix='/api/v1')
-app.register_blueprint(need_api, url_prefix='/api/v1')
-app.register_blueprint(response_api, url_prefix='/api/v1')
-app.register_blueprint(service_api, url_prefix='/api/v1')
-app.register_blueprint(user_api, url_prefix='/api/v1')
-
-queue.init_app(app)
-
 login_manager = LoginManager()
-login_manager.init_app(app)
-login_manager.login_view = 'login'
 
-csrf_protect.init_app(app)
 
-mailer.init_app(app)
+def create_app(cfg=dict()):
+    """
+    Setup an app with bluepirnts, extensions, and other necessary addtions.
+
+    cfg_dict: dict
+        dictionary to update app.config with, useful for testing.
+    """
+    # The app
+    app = Flask(__name__)
+
+    # Attempt to load config.py or fallback
+    try:
+        app.config.from_object('config')
+    except:
+        app.config.from_object('configdist')
+
+    # Additional configuration options (used by testing)
+    from _15thnight.database import init_db
+    ### OVERRIDE CONFIGURATION OPTIONS FOR TESTING HERE ###
```

Comment:
Intended to be noticed. Fixed up to be noticeable without such intensity for the next push.